Lib/test/support/__init__.py

```diff
@@ -1900,21 +1900,22 @@ def _run_suite(suite):
         raise TestFailed(err)
 
 
+_match_tests = None
+_match_test1 = None
+
 def _match_test(test):
-    global match_tests
+    global _match_tests, _match_test1
 
     if match_tests is None:
         return True
+    if match_tests != _match_tests:
+        match_tests_re = '|'.join(map(fnmatch.translate, match_tests))
+        _match_test1 = re.compile(match_tests_re, re.I).match
```

Comment:
Hum no, re.I must no be used: the match is expected to be **case sensitive**, not case insensitve.